src/main/java/com/solvd/laba/components/ProductItemComponent.java

```diff
@@ -0,0 +1,56 @@
+package com.solvd.laba.components;
+
+import java.time.Duration;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.pagefactory.DefaultElementLocatorFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.openqa.selenium.WebDriver;
+
+public class ProductItemComponent {
+
+    private final WebDriver driver;
+    private final WebDriverWait wait;
+
+    private static final Logger logger = LoggerFactory.getLogger(ProductItemComponent.class);
+
+    @FindBy(css = ".productinfo p")
```

Comment:
If the product name can change a lot, you can use parameterized XPath, which will search for an element based on part of the name.
.//div[contains(@class, 'productinfo')]//p[contains(text(), 'Blue Top')]

Reply:
In my current implementation, I chose to locate the product using a CSS selector (.productinfo p) and apply partial name matching via Java stream filtering. Specifically, I use the method findProductByPartialName(String partialName) which filters the product list using String.contains():
return getProductItems().stream()
        .filter(p -> p.getProductName().toLowerCase().contains(partialName.toLowerCase()))
        .findFirst()
        .orElseThrow(...);